### ISSUE TITLE
Laravel does not properly support MS SQL Table Schema

### ISSUE THREAD

**Issue by Mugluck** [laravel/framework#17353]
- Laravel Version: 5.3.28
- PHP Version: 7.0.13
- Database Driver & Version: Microsoft Azure SQL Server [ODBC Driver 13 for SQL Server]

### Description:
Attempting to run a many to many relation, or morph to many relation causes laravel 5.3 to incorrectly assign sql table schema (it simply drops them entirely) causing the SQL query to fail. This is likely caused by a lack of definition or place to define a schema name. 

Note: Calling a table directly in a model or code with work: "protected $table = 'schema_name.table';" results in a correctly formed sql query. 

However attempting

```
public function Model() {

      return  $this->morphedByMany('App\Model', 'polymorphictable');
    } 
```

```
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Polymorphictable extends Model
{
    protected $table = 'schema_name.polymorphictable'; 
 
    protected $fillable = ['Polymorphictarget_id','polymorphictable_id', 'model_type']; 
}

```
```
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Polymorphictarget extends Model
{
    protected $table = 'schema_name.Polymorphictarget'; 

    public function Model() {

      return  $this->morphedByMany('App\Model', 'polymorphictable');
    } 
}

```
Will produce a SQL query that looks like this: 

`SQLSTATE[42S02]: [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]Invalid object name 'polymorphictable'. (SQL: select [schema_name].[polymorphictarget].*, [polymorphictable].[polymorphictable_id] as [pivot_polymorphictable_id], [polymorphictable].[polymorphictarget_id] as [pivot_polymorphictarget_id] from [schema_name].[polymorphictarget] inner join [polymorphictable] on [schema_name].[polymorphictarget].[id] = [polymorphictable].[polymorphictarget_id] where [polymorphictable].[polymorphictable_id] in (196) and [polymorphictable].[model_type] = App\Model)`

As you can see, some of the query is properly formed, direct references to the table will properly reference the schema name, but the schema_name is not correctly included in other parts of the query. 

### Steps To Reproduce:

Create any polymorphic table and try to grab data with it. See code above. If you need a testing sql server set up, I can do this. Just message me and I'll take care of it.

**Comment by fernandobandeira** [comment]
What happens if you change
```php
      return  $this->morphedByMany('App\Model', 'polymorphictable');
```
to
```php
      return  $this->morphedByMany('App\Model', 'schema_name.polymorphictable');
```
? It should work...

**Comment by Mugluck** [comment]
I tried that, it only partially fills in the information. Some parts of the formed SQL query will properly reference the schema, while others will reference only the schema and not the table.

**Comment by thinkspill** [comment]
I just experienced an issue similar to this, and found this page during research. Here's what I found, for future visitors:

Background: Upgrading a Lumen 5.2 app using ODBC to Laravel 5.5.

Problem: Attempting any table query using ODBC with Laravel 5.5 would result in...

```
SQLSTATE[42S02]: Base table or view not found: 208 [Microsoft][ODBC Driver 13 for SQL Server]
[SQL Server]Invalid object name 'QueriedTableNameHere'.
```

Clues:

- Same .env values between Lumen and Laravel tests.
- Same query worked with ODBC under the Lumen branch only.
- Same query worked with sqlsrv under Lumen and Laravel.
- Same result when connected to local or staging DB instance.

Solution:

My odbc.ini file had some entries that were missing the `Database = dbname` line, so Laravel could see no default schema.

Apparently Lumen (5.2, anyway) is fine with this, but Laravel is not.

So if you're getting this strange message, check your odbc.ini.
